hail/python/hail/expr/matrix_type.py

```diff
@@ -0,0 +1,97 @@
+from hail.typecheck import *
+from hail.utils.java import escape_parsable
+from hail.expr.types import tstruct
+
+class tmatrix(object):
+    @typecheck_method(global_type=tstruct,
+                      col_type=tstruct, col_key=sequenceof(str),
+                      row_type=tstruct, row_key=sequenceof(str),
+                      entry_type=tstruct)
+    def __init__(self, global_type, col_type, col_key, row_type, row_key, entry_type):
+        self.global_type = global_type
+        self.col_type = col_type
+        self.col_key = col_key
+        self.row_type = row_type
+        self.row_key = row_key
+        self.entry_type = entry_type
+
+    def __eq__(self, other):
+        return (isinstance(other, ttable)
+                and self.global_type == other.global_type
+                and self.col_type == other.col_type
+                and self.col_key == other.col_key
+                and self.row_type == other.row_type
+                and self.row_key == other.row_key
+                and self.entry_type == other.entry_type)
+
+    def __ne__(self, other):
```

Comment:
Is this needed?

https://docs.python.org/3/reference/datamodel.html#object.__eq__
https://docs.python.org/3/reference/datamodel.html#object.__hash__

At least from those sources I don't see a condition where a user defined __ne__ is needed if it's the negation of __eq__, but I could easily be mistaken!

Reply:
yeah, I don't think we need this here

Reply:
Yeah, you're right.  From that link:

 > By default, __ne__() delegates to __eq__() and inverts the result unless it is NotImplemented.